depot/middleware.py

```diff
@@ -185,9 +185,12 @@ def __call__(self, environ, start_response):
                 full_path[mtpointlen:mtpointlen+1] in ('', '/')):
             return self.app(environ, start_response)
 
+        
+        mountpointlen=len(self.mountpoint.split('/'))
         path = full_path.split('/')
```

Comment:
I think that changing this `full_path.split('/')` to be `full_path.rsplit('/', 2)`  is the only change that we need to support subdirectories